### PR TITLE
Execute case

### PR DESCRIPTION
**Pull request by eivindlie**
Adds functionality to start and execute a new case. (Suspension is not handled ATM)

Probably some things that should be changed, but some input would be useful. Right now, the service itself spawns 5 worker processes that each check the database for any waiting cases. Perhaps it would make more sense to have a separate case worker container, and let Docker handle the scaling instead?

### CODE REVIEW

File: src/executor.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+from case_collection import CaseCollection
+from dotmap import DotMap
+from http import HTTPStatus
+import requests
+from multiprocessing import Process
+from case_collection import CaseStatus
+import time
```
Comment (sjrodahl) [inline src/executor.py]: PEP-ify the imports.

> Imports should be grouped in the following order:
 >  - standard library imports
 >  - related third party imports
 >  - local application/library specific imports
>
> You should put a blank line between each group of imports.

File: src/executor.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    :param endpoint: The endpoint that should be called on the WorkflowBlockService.
+    :param body: The body of the request as a JSON object.
+    :return: The response from the block.
+    """
+    url = block_url + endpoint
+    response = requests.post(url, json=body)
+    if response.status_code == HTTPStatus.OK:
+        return response.json()
+    else:
+        raise Exception('Endpoint {} returned status {}: {}'.format(endpoint, response.status_code, response.text))
```
Comment (sjrodahl) [inline src/executor.py]: Perhaps 'Endpoint {url} returned...'

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  case-executor-datastore:
+    image: mongo:3.0.2
+    networks:
+      - database
+  frontend:
+    build: .
+    ports:
+      - "5000:5000"
+    volumes:
+      - .:/src
```
Comment (sjrodahl) [inline docker-compose.yml]: add:
```
depends on:
    - case-executor-datastore
```

File: src/executor.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                    store = DotMap(case['store'])
+                    for saved_output, save_to in step_item['save_outputs'].items():
+                        if saved_output in result['data']:
+                            path = save_to.split('.')
+                            item = store
+                            for i in range(len(path) - 1):
+                                item = item[path[i]]
+                            item[path[-1]] = result['data'][saved_output]
+                    step = step_item['next_block']
+                    case['store'] = store.toDict()
```
Comment (sjrodahl) [inline src/executor.py]: Extract 128-139 to separate function for readability

File: src/executor.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                elif result['type'] == 'suspend':
+                    case['suspended'] = True
+                    # TODO Need to save state, and handle the reason for suspension
+                    pass
+
+            if step == '-1':
+                case['status'] = CaseStatus.FINISHED
```
Comment (sjrodahl) [inline src/executor.py]: This requirement for cases should be well documented

File: src/executor.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            time.sleep(5)
+        else:
+            execute_case(case)
+
+
+def start_workers():
+    for i in range(NUM_WORKERS):
+        worker = Process(target=worker_target)
+        workers.append(worker)
+        worker.start()
```
Comment (sjrodahl) [inline src/executor.py]: Agree with the notion to extract workers to a separate container and let docker handle scaling.

File: worker/src/worker.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+        while True:
+            step_item = workflow['blocks'][step]
+
+            if step_item['type'] == 'action':
+                print(step)
+                result = execute_block(case, step_item, step)
+
+                if not result:
+                    return
```
Comment (sjrodahl) [inline worker/src/worker.py:128]: Could a faulty block interrupt the entire workflow?

Comment (eivindlie) [reply worker/src/worker.py:128]: Yes, and the way I see it, this is also the correct behaviour. If a single block misbehaves, the following blocks will not have access to the expected data.

File: src/executor.py (hunk 5 above)
Comment (eivindlie) [reply src/executor.py]: Now documented in workflow-editor-service.

**Pull request closed without merging**